spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/cassandra/CassandraReactiveHealthIndicatorTests.java

```diff
@@ -59,19 +65,38 @@ void testCassandraIsUp() {
 
 	@Test
 	void testCassandraIsDown() {
-		ReactiveCassandraOperations reactiveCassandraOperations = mock(ReactiveCassandraOperations.class);
-		given(reactiveCassandraOperations.getReactiveCqlOperations())
-				.willThrow(new CassandraInternalException("Connection failed"));
+		CqlSession session = mock(CqlSession.class);
+		given(session.executeReactive(any(SimpleStatement.class)))
+				.willThrow(new DriverTimeoutException("Connection timed out (not really)"));
 
 		CassandraReactiveHealthIndicator cassandraReactiveHealthIndicator = new CassandraReactiveHealthIndicator(
-				reactiveCassandraOperations);
+				session);
 		Mono<Health> health = cassandraReactiveHealthIndicator.health();
 		StepVerifier.create(health).consumeNextWith((h) -> {
 			assertThat(h.getStatus()).isEqualTo(Status.DOWN);
 			assertThat(h.getDetails()).containsOnlyKeys("error");
 			assertThat(h.getDetails().get("error"))
-					.isEqualTo(CassandraInternalException.class.getName() + ": Connection failed");
+					.isEqualTo(DriverTimeoutException.class.getName() + ": Connection timed out (not really)");
 		}).verifyComplete();
 	}
 
+	private Answer<Void> mockReactiveResultSetBehavior(ReactiveRow row) {
```

Comment:
Meh. I am not keen to start mocking a `Publisher` here. The fact that the return type is a `Publisher` + something else makes it very hard to use existing reactor types. WIthout that I could have replaced that with `Mono.just`.

Reply:
Yeah, I don't see how to simplify this either. I'll take a note to provide a dummy implementation of `ReactiveResultSet` for test use cases like this one, in a future driver release.